sql/expression/function/greatest_least_test.go

```diff
@@ -0,0 +1,209 @@
+package function
+
+import (
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"testing"
+)
+
+func TestGreatest(t *testing.T) {
+	t.Run("null", func(t *testing.T) {
```

Comment:
Instead of repeating the same boilerplate over and over, what do you think about a table-driver test?

```go
func TestGreatest(t *testing.T) {
    testCases := []struct{
        name string
        args []sql.Expression
        expected interface{}
    }{
        {
            "null",
            []sql.Expression{
                expression.NewLiteral(nil, sql.Null),
			    expression.NewLiteral(5, sql.Int64),
			    expression.NewLiteral(1, sql.Int64),
            },
            nil,
        },
        {
            "negative and all ints",
            []sql.Expression{
                expression.NewLiteral(int64(-1), sql.Int64),
			    expression.NewLiteral(int64(5), sql.Int64),
			    expression.NewLiteral(int64(1), sql.Int64),
            },
            int64(5),
        },
        // ...
    }

    for _, tt := range testCases {
        t.Run(tt.name, func(t *testing.T) {
            require := require.New(t)

            f, err := NewGreatest(tt.args...)
            require.NoError(err)

            output, err := f.Eval(sql.NewEmptyContext(), nil)
            require.NoError(err)
            require.Equal(tt.expected, output)
        })
    }
}
```

Reply:
Done, thanks.